prover/src/prover.rs

```diff
@@ -0,0 +1,133 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+//
+// Copyright (c) DUSK NETWORK. All rights reserved.
+
+mod execute;
+mod stco;
+mod stct;
+mod wfco;
+mod wfct;
+
+use crate::{ProverError, ProverResult};
+
+use dusk_bytes::{DeserializableSlice, Serializable};
+use dusk_pki::PublicSpendKey;
+use dusk_plonk::prelude::Prover as PlonkProver;
+use once_cell::sync::Lazy;
+use rand::rngs::OsRng;
+
+use dusk_plonk::prelude::*;
+use dusk_schnorr::Signature;
+use dusk_wallet_core::UnprovenTransaction;
+use phoenix_core::{Crossover, Fee, Message};
+use rusk_profile::keys_for;
+
+use transfer_circuits::{
+    CircuitInput, CircuitInputSignature, DeriveKey, ExecuteCircuit,
+    SendToContractObfuscatedCircuit, SendToContractTransparentCircuit,
+    StcoCrossover, StcoMessage, WfoChange, WfoCommitment,
+    WithdrawFromObfuscatedCircuit, WithdrawFromTransparentCircuit,
+};
+
+pub use stco::STCO_INPUT_LEN;
+pub use stct::STCT_INPUT_LEN;
+pub use wfco::WFCO_INPUT_LEN;
+pub use wfct::WFCT_INPUT_LEN;
+
+/// Arity of the transfer tree.
+pub const A: usize = 4;
+
+#[derive(Debug, Default)]
+pub struct LocalProver {}
```

Comment:
Should this be `LocalProver {}` or `LocalProver`? Just checking, because as it is we can't instantiate it apart from with `Default`, and this instantiation could simply be `const`.